File: game-server/src/main/java/com/kingston/mmorpg/game/ai/fsm/State.java
package com.kingston.mmorpg.game.ai.fsm;

import com.kingston.mmorpg.game.scene.actor.Creature;

public interface State {

	/**
	 * 切换至新状态
	 * 
	 * @param creature
	 */
	void onEnter(Creature creature);

	/**
	 * 离开当前状态
	 * 
	 * @param creature
	 */
	void onExit(Creature creature);

	/**
	 * 每一个tick跑的业务
	 * 
	 * @param creature
	 */
	void execute(Creature creature);

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/fsm/State.kt
package org.forfun.mmorpg.game.ai.fsm

import org.forfun.mmmorpg.game.scene.actor.Creature


interface State {

    /**
     * 切换至新状态
     * @param creature
     */
    fun onEnter(creature: Creature)

    /**
     * 离开当前状态
     * @param creature
     */
    fun onExit(creature: Creature)

    /**
     * 每一个tick跑的业务
     * @param creature
     */
    fun execute(creature: Creature)
}
